GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -2126,6 +2136,35 @@ internal void ShowFirstParent()
             ForceRefreshRevisions();
         }
 
+        internal void ToggleBetweenArtificialAndHeadCommits()
+        {
+            GoToRef(GetIdToSelect()?.ToString(), showNoRevisionMsg: false);
+            ToggledBetweenArtificialAndHeadCommits?.Invoke(this, EventArgs.Empty);
+            return;
+
+            ObjectId GetIdToSelect()
+            {
+                // try the up to 3 possibilities
+                ObjectId idToSelect = LatestSelectedRevision?.ObjectId;
+                for (int i = 0; i < 3; ++i)
+                {
+                    idToSelect = GetNextIdToSelect(idToSelect);
+                    if (idToSelect != null
+                        && (!idToSelect.IsArtificial || !AppSettings.ShowGitStatusForArtificialCommits || GetChangeCount(idToSelect).HasChanges))
```

Comment:
Why return Artificial also for ShowGitStatusForArtificialCommits==false?

Reply:
> Why return Artificial also for ShowGitStatusForArtificialCommits==false?

There is no change-count information by hand. So do not just select HEAD but cycle through all three as the name of the function promises.

I'll add this to the issue description, when @NikolayXHD has explained his modification.

Reply:
added the comments and updated the issue description

Reply:
OK with the explanation, but not how I expect this to work

Reply:
So what did you expect?
If you want to run git commands (directly or indirectly), please tell me how to update `ChangeCount`.